stdlib/REPL/src/LineEdit.jl

```diff
@@ -432,12 +456,27 @@ prompt_string(p::Prompt) = prompt_string(p.prompt)
 prompt_string(s::AbstractString) = s
 prompt_string(f::Function) = Base.invokelatest(f)
 
+function maybe_show_hint(s::PromptState)
+    isa(s.hint, String) || return nothing
+    if isempty(s.hint)
+        print(terminal(s), "\e[0K")
```

Comment:
what does this do? I assume `s.hint == ""` means that a possible previous hint should be deleted? a short comment explaining this would be useful